### PR TITLE
Improve generation output path

### LINKED ISSUE #1130: gradle plugin generating code at the wrong folder
In my `app/build.gradle`
```
wire {
  kotlin {
    out "src/main/java"
  }
}
```
The generated code is going to `/src/main/java` instead of `/app/src/main/java`. I believe `WireCompiler` was doing it as expected.

- [ ] write tests.
- [ ] fix.

### PR DESCRIPTION
**Pull request by jrodbx**
Closes #1130

### CODE REVIEW

File: docs/wire_compiler.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ code. It uses [KotlinPoet][kotlinpoet] internally to generate compact and legibl
 
 ![Basic](images/gradle_basic@2x.png)
 
-Wire will generate files in `build/generated/src/main/java`. It'll also register this directory as a
+Wire will generate files in `build/generated/source/wire`. It'll also register this directory as a
```
Comment (jrodbx) [inline docs/wire_compiler.md]: kapt, apt use `build/generated/source`.
sqldelight use `build/sqldelight`.
the former seemed more conventional.

Comment (oldergod) [reply docs/wire_compiler.md]: I like `build/wire` more; isn't just beautifully simple?

Comment (jrodbx) [reply docs/wire_compiler.md]: It's shorter, for sure. 
On the other hand, buildd/generated/source/wire vs build/generated/source/apt vs build/generated/source/kapt has a nice symmetric structure IMO.  Do you feel strongly either way?

Comment (Egorand) [reply docs/wire_compiler.md]: My personal preference is for `build/generated/source/wire`, but both are fine IMO.

**Pull request merged**